City-of-Helsinki/drupal-helfi-platform-config
# UHF-8923 config ignore for chat and calculator

# [UHF-8923](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8923)
Added chats and calculators to config ignore alter hook

## What was done
Upgraded config ignore to 3.x
Added config filter as dependency (will be removed in future)
Removed config_filter from core.extensions

**implemented config_ignore alter hooks:** 
Added calculator settings to config ignore alter hook
Added chat block configurations to config ignore alter hook.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-8923 -W`
* Run `make drush-updb drush-cr`

## How to test
- run drush cim to make sure you have nothing extra to import
- Change chat and calculator config files from conf/cmi
- Run drush cim

**IBMChat and calculator configuration changes should not be imported.**
If calculator is not enabled in the environment: 
- run drush en helfi_calculator
- Run drush updb and drush cr and drush cex
- Change configuration files manually
- Run drush cim
- Ignored configuration should not change.

Chat leijuke:
- Edit chat leijuke translation configuration from conf/cmi/{langcode}
- drush cim
- Chat-leijuke configurations should not be imported


[UHF-8923]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8923?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
